### ISSUE TITLE
ClientAuthError: Nonce is not matching after several silent renewals

### ISSUE THREAD

**Issue by freeman-g** [AzureAD/microsoft-authentication-library-for-js#739]
## I'm submitting a...
- [x] Regression (a behavior that used to work and stopped working in a new release)
- [ ] Bug report  
- [ ] Performance issue
- [ ] Feature request
- [ ] Documentation issue or request
- [ ] Other... Please describe:
</code></pre>

## Browser:
- [x] Chrome version 74
- [ ] Firefox version XX
- [ ] IE version XX
- [ ] Edge version XX
- [ ] Safari version XX

## Library version

Library version: 1.0.1
Using the latest version

## Current behavior
We have a background renewal process to silently acquire new tokens. We have the renewal running 10 minutes before the tokens expire like this:

```
function scheduleBackgroundRenewal() {
  let jwt = parseJwt(window.user.idToken)
  let expiresAt = jwt.exp * 1000
  var delay = expiresAt - Date.now()

  // renew 10 minutes early
  delay -= 600000

  if (delay > 0) {
    window.tokenRenewalTimeout = setTimeout(() => {
      console.log('renewal timeout reached')
      renewTokens()
    }, delay)
  }
}

function renewTokens() {
  console.log('aquiring id token renewal')
  userAgentApplication.acquireTokenSilent({ scopes: msalGlobal.auth.graphIdTokenScopes })
    .then(idTokenResponse => {
      console.log('done with id token, aquiring access token renewal')
      userAgentApplication.acquireTokenSilent({ scopes: msalGlobal.auth.graphAccessTokenScopes }).then(accessTokenResponse => {
        setUser(idTokenResponse.idToken.rawIdToken, accessTokenResponse.accessToken)
        scheduleBackgroundRenewal()
      }).catch(error => {
        console.log(error)
      })
    }).catch(error => {
      console.log(error)
    })
}
```

We have received the following error message after the 4th renewal iteration:

```
main.js:159 renewal timeout reached
main.js:166 aquiring id token renewal
msal-global.js:34 Mon, 03 Jun 2019 14:38:18 GMT:00000-1.0.1-Verbose Calling addHint parameters
msal-global.js:34 Mon, 03 Jun 2019 14:38:18 GMT:00000-1.0.1-Verbose Token is not in cache for scope:<my app registration>
msal-global.js:34 Mon, 03 Jun 2019 14:38:18 GMT:00000-1.0.1-Verbose renewing idToken
msal-global.js:34 Mon, 03 Jun 2019 14:38:18 GMT:00000-1.0.1-Info renewidToken is called
msal-global.js:34 Mon, 03 Jun 2019 14:38:18 GMT:00000-1.0.1-Info Add msal frame to document:msalIdTokenFrame
msal-global.js:34 Mon, 03 Jun 2019 14:38:18 GMT:00000-1.0.1-Verbose Renew Idtoken Expected state: 1701ba7e-089b-445a-86f8-581fe6967b5e
msal-global.js:34 Mon, 03 Jun 2019 14:38:18 GMT:00000-1.0.1-Verbose Set loading state to pending for: 32cd4e95-22e9-460d-a675-8aee520d3274:1701ba7e-089b-445a-86f8-581fe6967b5e
msal-global.js:34 Mon, 03 Jun 2019 14:38:18 GMT:00000-1.0.1-Info LoadFrame: msalIdTokenFrame
msal-global.js:34 Mon, 03 Jun 2019 14:38:18 GMT:00000-1.0.1-Info Add msal frame to document:msalIdTokenFrame
msal-global.js:34 Mon, 03 Jun 2019 14:38:20 GMT:00000-1.0.1-Info Returned from redirect url
msal-global.js:34 Mon, 03 Jun 2019 14:38:20 GMT:00000-1.0.1-Info Processing the callback from redirect response
msal-global.js:34 Mon, 03 Jun 2019 14:38:20 GMT:00000-1.0.1-Info State status:true; Request type:RENEW_TOKEN
msal-global.js:34 Mon, 03 Jun 2019 14:38:20 GMT:00000-1.0.1-Info State is right
msal-global.js:34 Mon, 03 Jun 2019 14:38:20 GMT:00000-1.0.1-Info Fragment has id token
msal-global.js:34 Mon, 03 Jun 2019 14:38:20 GMT:00000-1.0.1-Error Nonce Mismatch.Expected Nonce: bc8078b1-0de3-45c8-801e-fd0c19677156,Actual Nonce: cdb15d88-f68d-40d7-9b8c-1d4e466e3d6b
main.js:196 ClientAuthError: Nonce is not matching, Nonce received:  bc8078b1-0de3-45c8-801e-fd0c19677156, nonce expected : cdb15d88-f68d-40d7-9b8c-1d4e466e3d6b.
```

It seems like there must be an issue with setting the nonce in the hidden iframe in certian silent renewal scenarios.

Also we suspect that it may be a timing issue and perhaps 10 minutes prior to expiration is not the best time to renew? We're not sure what timing forces msal to fetch new tokens vs. re-using existing tokens. We don't want to wait until the tokens are fully expired because we don't want to interrupt user flow.

## Expected behavior
We expect the tokens to renew silently.

## Minimal reproduction of the problem with instructions
This is not very easy to reproduce, but we can try if necessary.

</code></pre>

**Comment by DarylThayil** [comment]
@freeman-g  thanks for submitting the issue. We will definitely investigate.

@pkanher617  can you look at htis? 

@freeman-g  the library will refresh your token before it expires, so you may need the code here to do this yourself.
@pkanher617  can you add more color to this

**Comment by DarylThayil** [comment]
@pkanher617  do you have an update to add?

**Comment by pkanher617** [comment]
I will take a look at this today. Sorry for the delay.

**Comment by freeman-g** [comment]
Thank you for looking at this. For what its worth, I also noticed some timeout errors on the 4th silent renewal. I saw several other Github issues that's sounded similar related to acquireTokenSilent, so I didn't create a separate issue for that.

**Comment by pkanher617** [comment]
@freeman-g I pushed a fix for some of those other acquireTokenSilent issues to dev. Could you test it and let me know if it fixes this issue? We had a bug preventing async silent calls which seemed to be fixed for other use cases.

**Comment by freeman-g** [comment]
Yes, I suspected my issue was related to some of the other issues. I'll give the dev branch a try and let you know.  Thanks!

**Comment by freeman-g** [comment]
@pkanher617 Do you know how I can install the Dev branch with NPM? I tried this but got an error:

`npm install AzureAd/microsoft-authentication-library-for-js#dev --save`

And/or how to use the Dev branch? Do I need to do a typescript build?

Thanks

**Comment by pkanher617** [comment]
@freeman-g I have published the patches to the latest NPM. You should be able to test it from there.

In the future, you can use the following steps to run the dev branch (given by another user [here](https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/736#issuecomment-503711527))

1. Clone and build the repo locally.
2. Build the project. You will see the new dist files in msal-core/dist.
3. Run `npm uninstall msal` to remove the msal project from your node_modules.
4. Use `npm install file:<path-to-msal-core-folder>` to install the local version to your node_modules.

Ensure you run `npm uninstall msal` and `npm install msal` to get back to the latest.

Let me know if this works, I haven't been able to repro but I am trying some other things as well.

**Comment by freeman-g** [comment]
@pkanher617 I tested with MSAL 1.0.2 and I'm still having issues with acquireTokenSilent.

This is a little difficult to explain, so bear with me. 

- I ran my test in a new incognito tab with a fresh login and fresh local storage entries, etc
- I'm renewing both AccessTokens and ID tokens silently every 45 minutes via a Javascript setInterval loop.
- aquireTokenSilent gives me a bad token on the 4th renewal, so after 3 hours have gone by.
- I noticed that the aquireTokenSilent method with ID Token Scopes has a response that says tokenType: 'access_token' AND gives me back an expired token in the rawIdToken property. I think this is the core problem.
- See log file below

Here is the relevant code I'm using now:
```
function shouldRenewJwt(token) {
    let jwt = parseJwt(token)
    let expDate = new Date(jwt.exp * 1000) // convert unix timestamp to milliseconds
    let nearExpirationDate = new Date()
    nearExpirationDate.setMinutes(nearExpirationDate.getMinutes() + 15)
  
    // return true if the JWT is within 15 minutes of expiring
    if (expDate < nearExpirationDate) {
      return true
    }
    return false
  }

function startBackgroundRenewalLoop() {
  setInterval(() => {
    if (shouldRenewJwt(window.user.token)) {

      // get a new ID Token
      userAgentApplication.acquireTokenSilent({ scopes: msalGlobal.auth.graphIdTokenScopes })
        .then(idTokenResponse => {
          console.log('got id token', idTokenResponse)

          // get a new access token
          userAgentApplication.acquireTokenSilent({ scopes: msalGlobal.auth.graphAccessTokenScopes }).then(accessTokenResponse => {
            console.log('got access token', accessTokenResponse)

          }).catch(error => {
            console.log(error)
          })
        }).catch(error => {
          console.log(error)
        })
    }
  }, 1000)
}
```

Here is the relevant log. I've removed some of the log for security reasons but tried to leave the important parts:

```
// token initially acquired at 15:07
// first silent renewal
msal-global.js:34 Fri, 21 Jun 2019 15:52:38 GMT:00000-1.0.2-Verbose Calling addHint parameters
msal-global.js:34 Fri, 21 Jun 2019 15:52:38 GMT:00000-1.0.2-Info Token is already in cache for scope:<removed>
main.js:230 got id token {uniqueId: "", tenantId: "", tokenType: "id_token", idToken: e, accessToken: "<removed token>", …}

// at this point raw ID token expiration is 17:37

msal-global.js:34 Fri, 21 Jun 2019 15:52:38 GMT:00000-1.0.2-Verbose Calling addHint parameters
msal-global.js:34 Fri, 21 Jun 2019 15:52:38 GMT:00000-1.0.2-Info Token is already in cache for scope:https://graph.microsoft.com/user.read openid profile
main.js:234 got access token {uniqueId: "", tenantId: "", tokenType: "access_token", idToken: e, accessToken: "<removed token>", …}

// 2nd slient renewal
msal-global.js:34 Fri, 21 Jun 2019 16:37:39 GMT:00000-1.0.2-Verbose Calling addHint parameters
msal-global.js:34 Fri, 21 Jun 2019 16:37:39 GMT:00000-1.0.2-Verbose Token is not in cache for scope:<removed>
msal-global.js:34 Fri, 21 Jun 2019 16:37:39 GMT:00000-1.0.2-Verbose renewing idToken
msal-global.js:34 Fri, 21 Jun 2019 16:37:39 GMT:00000-1.0.2-Info renewidToken is called
msal-global.js:34 Fri, 21 Jun 2019 16:37:39 GMT:00000-1.0.2-Info Add msal frame to document:msalIdTokenFrame
msal-global.js:34 Fri, 21 Jun 2019 16:37:40 GMT:00000-1.0.2-Verbose Renew Idtoken Expected state: 5bee944e-737b-40f7-b4f6-09039c3e6710
msal-global.js:34 Fri, 21 Jun 2019 16:37:40 GMT:00000-1.0.2-Verbose Set loading state to pending for: <removed>:5bee944e-737b-40f7-b4f6-09039c3e6710
msal-global.js:34 Fri, 21 Jun 2019 16:37:40 GMT:00000-1.0.2-Info LoadFrame: msalIdTokenFrame
msal-global.js:34 Fri, 21 Jun 2019 16:37:40 GMT:00000-1.0.2-Info Add msal frame to document:msalIdTokenFrame
UserAgentApplication.js:834 Set-Cookie header is ignored in response from url: https://login.microsoftonline.com/<removed>. Cookie length should be less than or equal to 4096 characters.
(anonymous) @ UserAgentApplication.js:834
setTimeout (async)
e.loadFrame @ UserAgentApplication.js:831
e.loadIframeTimeout @ UserAgentApplication.js:807
e.renewIdToken @ UserAgentApplication.js:1389
(anonymous) @ UserAgentApplication.js:753
msal-global.js:34 Fri, 21 Jun 2019 16:37:43 GMT:00000-1.0.2-Info Returned from redirect url
msal-global.js:34 Fri, 21 Jun 2019 16:37:43 GMT:00000-1.0.2-Info Processing the callback from redirect response
msal-global.js:34 Fri, 21 Jun 2019 16:37:43 GMT:00000-1.0.2-Info State status:true; Request type:RENEW_TOKEN
msal-global.js:34 Fri, 21 Jun 2019 16:37:43 GMT:00000-1.0.2-Info State is right
msal-global.js:34 Fri, 21 Jun 2019 16:37:43 GMT:00000-1.0.2-Info Fragment has id token
msal-global.js:34 Fri, 21 Jun 2019 16:37:43 GMT:00000-1.0.2-Verbose acquiring token interactive in progress

// notice how the tokenType says "access_token" but I asked for ID token scopes
// the id token itself is not updated and still set to expire at 17:37
main.js:230 got id token {uniqueId: "a3cd903c-b3fe-4e1d-8665-b76a49da291b", tenantId: "<removed>", tokenType: "access_token", idToken: e, accessToken: null, …}
msal-global.js:34 Fri, 21 Jun 2019 16:37:43 GMT:00000-1.0.2-Verbose Calling addHint parameters
msal-global.js:34 Fri, 21 Jun 2019 16:37:43 GMT:00000-1.0.2-Verbose Token is not in cache for scope:https://graph.microsoft.com/user.read openid profile
msal-global.js:34 Fri, 21 Jun 2019 16:37:43 GMT:00000-1.0.2-Verbose renewing accesstoken
msal-global.js:34 Fri, 21 Jun 2019 16:37:43 GMT:00000-1.0.2-Verbose renewToken is called for scope:https://graph.microsoft.com/user.read openid profile
msal-global.js:34 Fri, 21 Jun 2019 16:37:43 GMT:00000-1.0.2-Info Add msal frame to document:msalRenewFramehttps://graph.microsoft.com/user.read openid profile
msal-global.js:34 Fri, 21 Jun 2019 16:37:43 GMT:00000-1.0.2-Verbose Renew token Expected state: 7b63aa8d-ab09-45f1-a297-322f6ed8a82f
msal-global.js:34 Fri, 21 Jun 2019 16:37:43 GMT:00000-1.0.2-Verbose Set loading state to pending for: https://graph.microsoft.com/user.read openid profile:7b63aa8d-ab09-45f1-a297-322f6ed8a82f
msal-global.js:34 Fri, 21 Jun 2019 16:37:43 GMT:00000-1.0.2-Info LoadFrame: msalRenewFramehttps://graph.microsoft.com/user.read openid profile
msal-global.js:34 Fri, 21 Jun 2019 16:37:44 GMT:00000-1.0.2-Info Add msal frame to document:msalRenewFramehttps://graph.microsoft.com/user.read openid profile
UserAgentApplication.js:834 Set-Cookie header is ignored in response from url:  https://login.microsoftonline.com/<removed>. Cookie length should be less than or equal to 4096 characters.
(anonymous) @ UserAgentApplication.js:834
setTimeout (async)
e.loadFrame @ UserAgentApplication.js:831
e.loadIframeTimeout @ UserAgentApplication.js:807
e.renewToken @ UserAgentApplication.js:1363
(anonymous) @ UserAgentApplication.js:758
msal-global.js:34 Fri, 21 Jun 2019 16:37:45 GMT:00000-1.0.2-Info Returned from redirect url
msal-global.js:34 Fri, 21 Jun 2019 16:37:45 GMT:00000-1.0.2-Info Processing the callback from redirect response
msal-global.js:34 Fri, 21 Jun 2019 16:37:45 GMT:00000-1.0.2-Info State status:true; Request type:RENEW_TOKEN
msal-global.js:34 Fri, 21 Jun 2019 16:37:45 GMT:00000-1.0.2-Info State is right
msal-global.js:34 Fri, 21 Jun 2019 16:37:45 GMT:00000-1.0.2-Info Fragment has access token
msal-global.js:34 Fri, 21 Jun 2019 16:37:45 GMT:00000-1.0.2-Info The user object received in the response is the same as the one passed in the acquireToken request
msal-global.js:34 Fri, 21 Jun 2019 16:37:45 GMT:00000-1.0.2-Verbose acquiring token interactive in progress
main.js:234 got access token {uniqueId: "a3cd903c-b3fe-4e1d-8665-b76a49da291b", tenantId: "<removed>", tokenType: "access_token", idToken: e, accessToken: "<removed token>", …}


// starting 3rd silent renewal
msal-global.js:34 Fri, 21 Jun 2019 17:22:45 GMT:00000-1.0.2-Verbose Calling addHint parameters
msal-global.js:34 Fri, 21 Jun 2019 17:22:45 GMT:00000-1.0.2-Info Token is already in cache for scope:openid profile

// notice how the tokenType says "access_token" but I asked for ID token scopes
// the id token itself is not updated and still set to expire at 17:37
main.js:230 got id token {uniqueId: "", tenantId: "", tokenType: "access_token", idToken: e, accessToken: "<removed token>", …}
msal-global.js:34 Fri, 21 Jun 2019 17:22:45 GMT:00000-1.0.2-Verbose Calling addHint parameters
msal-global.js:34 Fri, 21 Jun 2019 17:22:45 GMT:00000-1.0.2-Info Token is already in cache for scope:https://graph.microsoft.com/user.read openid profile
main.js:234 got access token {uniqueId: "", tenantId: "", tokenType: "access_token", idToken: e, accessToken: "<removed token>", …}

// starting 4th silent renewal
msal-global.js:34 Fri, 21 Jun 2019 18:07:46 GMT:00000-1.0.2-Verbose Calling addHint parameters
msal-global.js:34 Fri, 21 Jun 2019 18:07:46 GMT:00000-1.0.2-Verbose Token is not in cache for scope:openid profile
msal-global.js:34 Fri, 21 Jun 2019 18:07:46 GMT:00000-1.0.2-Verbose renewing accesstoken
msal-global.js:34 Fri, 21 Jun 2019 18:07:46 GMT:00000-1.0.2-Verbose renewToken is called for scope:openid profile
msal-global.js:34 Fri, 21 Jun 2019 18:07:46 GMT:00000-1.0.2-Info Add msal frame to document:msalRenewFrameopenid profile
msal-global.js:34 Fri, 21 Jun 2019 18:07:47 GMT:00000-1.0.2-Verbose Renew token Expected state: 882d5cfc-0bf1-46f3-99ed-342c822c79c6
msal-global.js:34 Fri, 21 Jun 2019 18:07:47 GMT:00000-1.0.2-Verbose Set loading state to pending for: openid profile:882d5cfc-0bf1-46f3-99ed-342c822c79c6
msal-global.js:34 Fri, 21 Jun 2019 18:07:47 GMT:00000-1.0.2-Info LoadFrame: msalRenewFrameopenid profile
msal-global.js:34 Fri, 21 Jun 2019 18:07:47 GMT:00000-1.0.2-Info Add msal frame to document:msalRenewFrameopenid profile
UserAgentApplication.js:834 Set-Cookie header is ignored in response from url:  https://login.microsoftonline.com/<removed>. Cookie length should be less than or equal to 4096 characters.
(anonymous) @ UserAgentApplication.js:834
setTimeout (async)
e.loadFrame @ UserAgentApplication.js:831
e.loadIframeTimeout @ UserAgentApplication.js:807
e.renewToken @ UserAgentApplication.js:1363
(anonymous) @ UserAgentApplication.js:758
msal-global.js:34 Fri, 21 Jun 2019 18:07:50 GMT:00000-1.0.2-Info Returned from redirect url
msal-global.js:34 Fri, 21 Jun 2019 18:07:50 GMT:00000-1.0.2-Info Processing the callback from redirect response
msal-global.js:34 Fri, 21 Jun 2019 18:07:50 GMT:00000-1.0.2-Info State status:true; Request type:RENEW_TOKEN
msal-global.js:34 Fri, 21 Jun 2019 18:07:50 GMT:00000-1.0.2-Info State is right
msal-global.js:34 Fri, 21 Jun 2019 18:07:50 GMT:00000-1.0.2-Info Fragment has access token
msal-global.js:34 Fri, 21 Jun 2019 18:07:50 GMT:00000-1.0.2-Info The user object received in the response is the same as the one passed in the acquireToken request
msal-global.js:34 Fri, 21 Jun 2019 18:07:50 GMT:00000-1.0.2-Verbose acquiring token interactive in progress

// notice how the tokenType says "access_token" but I asked for ID token scopes
// the id token itself is not updated and still set to expire at 17:37
main.js:230 got id token {uniqueId: "a3cd903c-b3fe-4e1d-8665-b76a49da291b", tenantId: "<removed>", tokenType: "access_token", idToken: e, accessToken: "<removed>", …}
msal-global.js:34 Fri, 21 Jun 2019 18:07:50 GMT:00000-1.0.2-Verbose Calling addHint parameters
msal-global.js:34 Fri, 21 Jun 2019 18:07:50 GMT:00000-1.0.2-Verbose Token is not in cache for scope:https://graph.microsoft.com/user.read openid profile
msal-global.js:34 Fri, 21 Jun 2019 18:07:50 GMT:00000-1.0.2-Verbose renewing accesstoken
msal-global.js:34 Fri, 21 Jun 2019 18:07:50 GMT:00000-1.0.2-Verbose renewToken is called for scope:https://graph.microsoft.com/user.read openid profile
msal-global.js:34 Fri, 21 Jun 2019 18:07:50 GMT:00000-1.0.2-Info Add msal frame to document:msalRenewFramehttps://graph.microsoft.com/user.read openid profile
msal-global.js:34 Fri, 21 Jun 2019 18:07:50 GMT:00000-1.0.2-Verbose Renew token Expected state: 124f31b5-408e-44c7-bbaa-749a5dd03ab8
msal-global.js:34 Fri, 21 Jun 2019 18:07:50 GMT:00000-1.0.2-Verbose Set loading state to pending for: https://graph.microsoft.com/user.read openid profile:124f31b5-408e-44c7-bbaa-749a5dd03ab8
msal-global.js:34 Fri, 21 Jun 2019 18:07:50 GMT:00000-1.0.2-Info LoadFrame: msalRenewFramehttps://graph.microsoft.com/user.read openid profile
msal-global.js:34 Fri, 21 Jun 2019 18:07:51 GMT:00000-1.0.2-Info Add msal frame to document:msalRenewFramehttps://graph.microsoft.com/user.read openid profile
UserAgentApplication.js:834 Set-Cookie header is ignored in response from url: https://login.microsoftonline.com/<removed>. Cookie length should be less than or equal to 4096 characters.
(anonymous) @ UserAgentApplication.js:834
setTimeout (async)
e.loadFrame @ UserAgentApplication.js:831
e.loadIframeTimeout @ UserAgentApplication.js:807
e.renewToken @ UserAgentApplication.js:1363
(anonymous) @ UserAgentApplication.js:758
msal-global.js:34 Fri, 21 Jun 2019 18:07:52 GMT:00000-1.0.2-Info Returned from redirect url
msal-global.js:34 Fri, 21 Jun 2019 18:07:52 GMT:00000-1.0.2-Info Processing the callback from redirect response
msal-global.js:34 Fri, 21 Jun 2019 18:07:52 GMT:00000-1.0.2-Info State status:true; Request type:RENEW_TOKEN
msal-global.js:34 Fri, 21 Jun 2019 18:07:52 GMT:00000-1.0.2-Info State is right
msal-global.js:34 Fri, 21 Jun 2019 18:07:52 GMT:00000-1.0.2-Info Fragment has access token
msal-global.js:34 Fri, 21 Jun 2019 18:07:52 GMT:00000-1.0.2-Info The user object received in the response is the same as the one passed in the acquireToken request
msal-global.js:34 Fri, 21 Jun 2019 18:07:52 GMT:00000-1.0.2-Verbose acquiring token interactive in progress
main.js:234 got access token {uniqueId: "a3cd903c-b3fe-4e1d-8665-b76a49da291b", tenantId: "<removed>", tokenType: "access_token", idToken: e, accessToken: "<token removed>", …}

// at this point the ID token returned from Get ID Token is already expired
// calls to my backend now fail
ajax.ts:153 POST https://mybackend.server.com/api/auth/test 401 (Unauthorized)
```

**Comment by freeman-g** [comment]
I think my issue may be the same as https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/759.

I'll be glad to test the fix for 759 whenever you are ready.

Thanks

**Comment by olkuziv** [comment]
Guys, I have the same issue, someone fixes it, or how I can fix it locally??

**Comment by DarylThayil** [comment]
Adding this to our October 2019 milestone to resolve

**Comment by JosephThomas** [comment]
Hey team,

I manually built the repo with npm test and installed it into my repo as mentioned by pkanher617, however the issue with asynchronous silent token acquiring seems to persist, with the ClientAuthError: Nonce is not matching. The issue fixes itself on a refresh (since the tokens are already in cache, MSAL doesn't seem to check any nonce if it finds them in cache).  I also noticed  pkanher617 saying that they had published the patch to the latest NPM, so should the dev branch have a fix or a version of MSAL? I tried both but still getting the same issue. Any guidance is appreciated.

Thanks,
Joseph
